application/controllers/Admin.php

```diff
@@ -55,6 +56,87 @@ public function markAsPaid() {
         }
     }
 
+    /**
+     * Checks an input key against a key stored in a file. If it matches, store a cookie on the users browser.
+     */
+    public function authenticate() {
+
+        //gets the passkey from private_keys/auth_props.json
+        if (!defined('Passkey')) {
+            $json = file_get_contents("private_keys/auth_props.json");
+            $authProps = json_decode($json, true);
+        
+            define('Passkey', $authProps["auth.passkey"]);
+        }
+        // Scrambler key
+        $jwtKey = 'JWTKEY';
+
+        // Gets a key from the URL from the form admin/authenticate?key=xyz
+        $urlKey = $this->input->get('key');
+        
+        // Check if it matches a key we have stored
+        if ($urlKey != Passkey) {
+            echo("Key is incorrect");
+            return false;
+        }
+
+        $cookieName = 'admin_authentication';
```

Comment:
Rename the cookie to be specific around ASPA, and move this to the outside of this function (just inside the admin class), like this:

`const AUTH_COOKIE_NAME = "aspa_admin_authentication";`

to access this, you can use `self::AUTH_COOKIE_NAME`

This way, we don't need to redefine it in the `checkCookie()` method